applications/amathea/src/features/map/Map.tsx

```diff
@@ -1,59 +1,42 @@
-import React, { useEffect, useState, useCallback, useMemo } from 'react'
+import React, { useEffect, useState, useCallback } from 'react'
 import GFWAPI from '@globalfishingwatch/api-client'
 import { InteractiveMap, MapRequest } from '@globalfishingwatch/react-map-gl'
 import Miniglobe, { MiniglobeBounds } from '@globalfishingwatch/ui-components/dist/miniglobe'
 import MapLegend from '@globalfishingwatch/ui-components/dist/map-legend'
 import IconButton from '@globalfishingwatch/ui-components/dist/icon-button'
 import useLayerComposer from '@globalfishingwatch/react-hooks/dist/use-layer-composer'
-import { ExtendedLayer, ExtendedStyle } from '@globalfishingwatch/layer-composer/dist/types'
+import {
+  useMapHover,
+  InteractionEventCallback,
+} from '@globalfishingwatch/react-hooks/dist/use-map-interaction'
+import useMapLegend from '@globalfishingwatch/react-hooks/dist/use-map-legend'
 import { useUserConnect } from 'features/user/user.hook'
+import { useDataviewsConnect } from 'features/dataviews/dataviews.hook'
 import { useMapboxRef } from './map.context'
 import { useGeneratorsConnect, useViewport } from './map.hooks'
 import styles from './Map.module.css'
 
 import '@globalfishingwatch/mapbox-gl/dist/mapbox-gl.css'
 
-type LegendConfig = Record<string, number>
-
-// TODO: move this to its own component?
-function useLegendComposer(style: ExtendedStyle, currentValues: LegendConfig) {
-  const layersWithLegend = useMemo(() => {
-    return style
-      ? style?.layers
-          ?.flatMap((layer) => {
-            if (!layer.metadata?.legend) return []
-
-            return {
-              ...layer,
-              ...(currentValues[layer.id] !== undefined && {
-                metadata: {
-                  ...layer.metadata,
-                  legend: { ...layer.metadata.legend, currentValue: currentValues[layer.id] },
-                },
-              }),
-            }
-          })
-          // Reverse again to keep dataviews sidebar and legend aligned
-          .reverse()
-      : ([] as ExtendedLayer[])
-  }, [currentValues, style])
-  return layersWithLegend
-}
-
 const Map = (): React.ReactElement => {
   const mapRef = useMapboxRef()
   const { viewport, onViewportChange, setMapCoordinates } = useViewport()
   const { latitude, longitude, zoom } = viewport
   const { logged } = useUserConnect()
+  const { dataviewsList } = useDataviewsConnect()
```

Comment:
Now I'm having some doubts about needing this everywhere... why should map know about dataviews?
maybe it is easier to merge at the beginning of the chain, aka layer-composer, and grab every information from the `style` output but let's discuss next week